pkg/core/mpt/node.go

```diff
@@ -0,0 +1,68 @@
+package mpt
+
+import (
+	"fmt"
+
+	"github.com/nspcc-dev/neo-go/pkg/io"
+	"github.com/nspcc-dev/neo-go/pkg/util"
+)
+
+// NodeType represents node type..
+type NodeType byte
+
+// Node types definitions.
+const (
+	BranchT    NodeType = 0x00
+	ExtensionT NodeType = 0x01
+	HashT      NodeType = 0x02
+	LeafT      NodeType = 0x03
+)
+
+// NodeWithType represents Node together with it's type.
+type NodeWithType struct {
```

Comment:
`NodeObj` or `AbstractNode`? It lost its `Type`, so we need some other way to differentiate it from `Node` interface. And it's not used a lot, just to serialize/deserialize things, so we can afford longer type name here.

Reply:
Fixed.